overlord/snapstate/snapstate.go

```diff
@@ -66,49 +66,53 @@ func doInstall(s *state.State, curActive bool, ss *SnapSetup) (*state.TaskSet, e
 		ss.Channel = "stable"
 	}
 
-	var prepare *state.Task
-	if ss.SnapPath != "" {
+	var prepare, prev *state.Task
```

Comment:
given the changes to doInstall maybe we need a test in the style of TestInstallTasks like TestRevertTasks, to make sure the chaining is right